include/nighthawk/common/sink.h

```diff
@@ -0,0 +1,46 @@
+
+#pragma once
+
+#include <vector>
+
+#include "envoy/common/pure.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+
+#include "api/client/service.grpc.pb.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Nighthawk {
+
+/**
+ * Abstract Sink interface.
+ */
+class Sink {
+public:
+  virtual ~Sink() = default;
+
+  /**
+   * Store an ExecutionResponse instance. Can be called multiple times for the same execution_id to
+   * persist multiple fragments that together will represent results belonging to a single
+   * execution.
+   *
+   * @param response Specify an ExecutionResponse instance that should be persisted. The
+   * ExecutionResponse must have its execution_id set.
+   * @return absl::Status Indicates if the operation succeeded or not.
+   */
+  virtual absl::Status
+  StoreExecutionResultPiece(const ::nighthawk::client::ExecutionResponse& response) const PURE;
+
+  /**
+   * Attempt to load a vector of ExecutionResponse instances associated to an execution id.
+   *
+   * @param execution_id Specify an execution_id that the desired set of ExecutionResponse
+   * instances are tagged with.
+   * @return const absl::StatusOr<std::vector<::nighthawk::client::ExecutionResponse>>
+   */
+  virtual const absl::StatusOr<std::vector<::nighthawk::client::ExecutionResponse>>
```

Comment:
Why are we enforcing the const-ness of the return value? This seems to make a copy, is it important to prevent the caller from performing modifications?

Reply:
No need for that, changed in 8f8a5da